File: py/graph_main.py
import sys
import json
from graph import Graph, Node
from graph_dfs import Graph_Searching_DFS
from graph_bfs import Graph_Searching_BFS


if __name__ == '__main__':

	argv = sys.argv
	ins = argv[1:]

	data_dir = {
		"directed": "../data/directed.json",
		"undirected": "../data/undirected.json",
		"dag": "../data/dag.json"
	}

	if ins[0] == "create_graph":
		pass 
	elif ins[0] == "search_graph":
		start_id = ins[3]
		end_id = ins[4]
	else:
		raise Exception('Wrong file argument')

	target = data_dir.get(ins[1])
	if not target:
		raise Exception('Wrong file argument')
	
	# initialized graph
	graph_obj = Graph()
	for _ in range(int(input())):
		graph_obj.add_node(Node(*input().split(' ')))


	for _ in range(int(input())):
		source, destination = input().split(' ')

		if ins[1] == 'undirected':
			graph_obj.add_undirected_edges(sourId=source, 
				destId=destination)
		else:
			graph_obj.add_directed_edges(sourId=source, 
				destId=destination)

	# run instruction
	if ins[0] == "create_graph":
		output = {
			'nodes': graph_obj.generate_nodes_list(),
			'edges': graph_obj.generate_all_edges_list()
		}

		if ins[1] == 'dag':
			if len(graph_obj.generate_topo_sort()) != len(graph_obj.nodes):
				raise Exception('Not a DAG data')

		with open(target, 'w') as f:
			json.dump(output, f, indent=3)

	elif ins[0] == "search_graph":
		if ins[2] == "bfs":
			search_obj = Graph_Searching_BFS(graph_obj)
		elif ins[2] == "dfs":
			search_obj = Graph_Searching_DFS(graph_obj)
		else:
			raise Exception("Option not available")
		result = search_obj.find_the_node(start_id, end_id)
		print(result)

File: js/graph_main.js
const { Graph, Node } = require("./graph.js");
const Graph_Searching_DFS = require("./graph_dfs.js");
const Graph_Searching_BFS = require("./graph_bfs.js");
const Graph_Djisktra = require("./graph_djikstra.js");
const input = require("readline-sync");
const fs = require("fs");

let argv = process.argv.slice(2);
let ins = argv;

let data_dir = {
  directed: "../data/directed.json",
  undirected: "../data/undirected.json",
  dag: "../data/dag.json",
};

var start_id, end_id;
if (ins[0] === "create_graph") {
  // pass
} else if (ins[0] === "search_graph") {
  start_id = ins[3];
  end_id = ins[4];
} else if (ins[0] === "shortest_path") {
  start_id = ins[2];
  end_id = ins[3];
} else {
  throw Error("Wrong file argument");
}

let target = data_dir[ins[1]];
if (target === undefined) {
  throw Error("Wrong file argument");
}

// initialized graph
const graph_obj = new Graph();
let n = input.question();
var curNode;

for (let i = 0; i < n; i++) {
  curNode = new Node(...input.question().split(" "));
  graph_obj.add_node(curNode);
}

n = input.question();
var curEdg;

for (let i = 0; i < n; i++) {
  curEdg = input.question().split(" ");

  if (ins[1] === "undirected") {
    graph_obj.add_undirected_edges(curEdg[0], curEdg[1], curEdg[2]);
  } else {
    graph_obj.add_directed_edges(curEdg[0], curEdg[1], curEdg[2]);
  }
}

// run instruction
if (ins[0] === "create_graph") {
  let output = {
    nodes: graph_obj.generate_nodes_list(),
    edges: graph_obj.generate_all_edges_list(),
  };

  if (ins[1] === "dag") {
    if (
      graph_obj.generate_topo_sort().length !==
      Object.keys(graph_obj.nodes).length
    ) {
      throw new Error("Not a DAG data");
    }
  }

  fs.writeFileSync(target, JSON.stringify(output, null, 3));
} else if (ins[0] === "search_graph") {
  var search_obj;
  if (ins[2] === "bfs") {
    search_obj = new Graph_Searching_BFS(graph_obj);
  } else if (ins[2] === "dfs") {
    search_obj = new Graph_Searching_DFS(graph_obj);
  } else {
    throw new Error("Option not available");
  }
  let result = search_obj.find_the_node(start_id, end_id);
  console.log(result);
} else if (ins[0] === "shortest_path") {
  let search_obj = new Graph_Djisktra(graph_obj);
  let result = search_obj.search(start_id, end_id);
  console.log(result);
}
